test/e2e/swaps/swaps-notifications.spec.js

```diff
@@ -111,7 +111,7 @@ describe('Swaps - notifications', function () {
         });
         await checkNotification(driver, {
           title: 'Insufficient balance',
-          text: 'You need 50 more TESTETH to complete this swap',
+          text: 'You need 43.4467 more TESTETH to complete this swap',
```

Comment:
@dan437 do you know why this number needed to change?

Reply:
For some reason it was ignoring test user's balance before. E.g. if a test user had a balance of 6.5533 TESTETH and wanted to swap 50 TESTETH, we were saying that 50 more is needed, even though it should have been 43.4467 TESTETH.

Reply:
Maybe [this PR](https://github.com/MetaMask/metamask-extension/pull/20370) is no longer needed if it was fixed here.

Reply:
I'm looking into this, will report on findings

Reply:
I think it is because the TESTETH token is being added to state as a token, and so then the `tokenBalance` variable in `review-quote.js` gets defined, and then `tokenBalanceNeeded` gets set, and then that is the value used to create `actionableBalanceErrorMessage`. Whereas previous to this PR, testeth was not being added to state as a token and so `tokenBalance` was falsy. Meanwhile, `ethBalanceNeeded` in `review-quote.js` is 0 when it should not be (I think this is unrelated to this PR).

We need to determine whether there are unintended consequences of this change to how tokenBalance works in `review-quote.js`. It is possible that we should not add a token if it is the main network token.

Reply:
oh, wait, I think I got that wrong...

Reply:
partially wrong: prior to this PR `ethBalanceNeeded` is correct, but it wasn't being used prior to this PR because `tokenBalanceNeeded` is truthy

why is that?

Reply:
Okay, I think I've made sense of it. This code would previously default to 0 if the user had no tokens added:
```
  const tokenBalance =
    tokensWithBalances?.length &&
    calcTokenAmount(
      selectedFromToken.balance || '0x0',
      selectedFromToken.decimals,
    ).toFixed(9);
```

The PR changes things so that the "to token" is now added to state, so `tokensWithBalances` has length, and the above `calcTokenAmount` function is called. We perhaps should have based this logic on the truthy-ness of `selectedFromToken.balance` and not `tokenWithBalances?.length`, but that is an issue that predates this PR.

The test previously asserted an incorrect value, and now this change in the PR causes the value to be correct in this instance. However, it seems possible that if the "to token" is the `swapsDefaultToken`, then the problem of the incorrect balance needed rendered here will persist (when there are no other tokens in state), because in that case the "to token" won't be added to state. Again, that problem predates this PR.

`tokenBalance` is only used for `tokenBalanceNeeded` and to pass to the modal for editing approval permissions. In both cases, the state of affairs in this PR seems fine, even if fixing `tokenBalanceNeeded` was an unintended consequence of this PR.

Importantly, this part is not true "I think it is because the TESTETH token is being added to state as a token", which is what I was worried about. It is just the "to token" (USDC in the case of this test). So I believe that the test change in this PR, and the behaviour change that necessitated the test change, are safe. Meanwhile, there are some other things that need to be cleaned up as alluded to above, but that should be done separately from this PR.

Reply:
> Maybe [this PR](https://github.com/MetaMask/metamask-extension/pull/20370) is no longer needed if it was fixed here.

Confirming that this PR fixed the issue #20370 was addressing